site/documentation/content/concepts/device-identity.md

```diff
@@ -147,3 +147,34 @@ After having verified the client certificate using the trust anchor(s), the prot
 the client certificate's *subject DN* and invokes the Credentials API's
 [*get Credentials*]({{< relref "/api/credentials#get-credentials" >}}) operation in order to retrieve the
 [*x509-cert* type credentials]({{< relref "/api/credentials#x509-certificate" >}}) that are on record for the device.
+
+### JSON Web Token based Authentication
+
+The MQTT protocol adapter supports authentication of devices with a JSON Web Token (JWT) based mechanism. In this case,
+the protocol adapter verifies the JWT by using the public key / certificate that is on record for the device in the
```

Comment:
it is just the public key FMPOV

Reply:
The mechanism also supports the possibility to use a X509 certificate for validation. In this case it uses the public key within the certificate to validate the JWT.

Reply:
Agreed, my point is that it always drills down to the public key being required for verifying the JWT's signature. We can therefore simply store the (encoded) public key in the registry and do not need to store an X.509 certificate ...

Reply:
Google IoT Core provides the possibility to verify device public key certificates against registry (tenant)-level CA certificates: https://cloud.google.com/iot/docs/how-tos/credentials/verifying-credentials
I would like to keep this option open for the future by storing also the certificate.

Reply:
I was not able to figure out, in which context a device would present its certificate, i.e. as part of the TLS handshake or otherwise. In any case, Hono already supports management of trust anchors (CA certificates) at the tenant level. This means that you can store CA certs for a tenant and then use these trust anchors for validating an X.509 certificate presented by a device.

Reply:
I think this is a slight misunderstanding, the device itself will not present the certificate. The possibility Google IoT Core provides is to verify a certificate against a registry (tenant)-level CA during the registration of the device's credentials (the certificate) with the device management API.

Reply:
IMHO you are talking about a method to make sure that keys for devices can only be registered by means of uploading a device certificate that has been signed by a root CA registered for the tenant. Hono currently has no such functionality but it could be added, of course. In any case, the Device Registry Management API would be used for doing so. Such a check could be done when a device is being provisioned with a certificate. However, for authenticating the device later on, it is sufficient to only store the public key.

Reply:
Exactly. Is there any class which we could use to extract the public key from the certificate? We found the class "TrustedCertificateAuthority" but this seems to only be for the tenant CA context, am I right in this assumption?